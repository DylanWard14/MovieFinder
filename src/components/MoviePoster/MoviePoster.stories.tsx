import React from 'react';
import { withKnobs, text } from '@storybook/addon-knobs';

import MoviePoster from './MoviePoster';

export default {
    title: 'Components/MoviePoster',
    decorators: [withKnobs],
};

export const DefaultExample = () => <MoviePoster imageURLExtension="/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg" />;

export const PropsExample = () => (
    <MoviePoster imageURLExtension={text('URL Extenstion', '/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg')} />
);
