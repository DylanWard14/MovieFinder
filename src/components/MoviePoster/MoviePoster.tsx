import React from 'react';
import styled from 'styled-components/macro';

interface MoviePosterProps {
    imageURLExtension: string;
}

const Poster = styled.img.attrs<{url: string}>({
    
})`
    width: 100%;
    height: 100%;
`;

const MoviePoster: React.FunctionComponent<MoviePosterProps> = ({imageURLExtension}) => {
    const fullImageUrl = `https://image.tmdb.org/t/p/original${imageURLExtension}`;
    return <Poster src={fullImageUrl}/>;
}

export default MoviePoster;