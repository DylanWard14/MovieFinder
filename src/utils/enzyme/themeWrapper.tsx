import React from 'react';
import { render, shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';

export const renderWrapper = (component: React.ReactNode) => {
    return render(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const shallowWrapper = (component: React.ReactNode) => {
    return shallow(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const mountWrapper = (component: React.ReactNode) => {
    return mount(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};
