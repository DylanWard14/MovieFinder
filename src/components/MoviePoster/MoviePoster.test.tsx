import React from 'react';
import { render } from 'enzyme';

import MoviePoster from './MoviePoster';

describe('MoviePoster', () => {
    it('renders the movie poster correctly', () => {
        const wrapper = render(<MoviePoster imageURLExtension="/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg" />);
        expect(wrapper).toMatchSnapshot();
    });
});
