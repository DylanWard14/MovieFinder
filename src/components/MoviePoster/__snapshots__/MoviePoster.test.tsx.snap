// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePoster renders the movie poster correctly 1`] = `
<img
  class="MoviePoster__Poster-nu5x18-0 bxbeZu"
  src="https://image.tmdb.org/t/p/original/btTdmkgIvOi0FFip1sPuZI2oQG6.jpg"
/>
`;
